Site Export fails in web farms when file replication (DFS) locks the zip packages during the process

## Description of bug
When sites are set up in a webfarm with file system replication (DFS), the site Export process finds the files locked and fails. Actually, it doesn't show an error on screen. It simply skips the problematic file. In the end, the exported files count is lower than it should be.

## Steps to reproduce
1. Set up a farm with file system replication across nodes
2. Perform a full export of a site

## Current behavior
The export process finishes but the following exception is shown in the logs:
```
2020-05-28 06:15:36.336-07:00 [DNNWD31RTSB2-02][D:22][T:26][ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.IO.IOException: The process cannot access the file '\\dnn.local\dnndfs\dnn123456-sandbox\www\App_Data\ExportImport\2020-05-28_11-14-59\export_packages.zip' because it is being used by another process.
at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
at System.IO.Compression.ZipFile.Open(String archiveFileName, ZipArchiveMode mode, Encoding entryNameEncoding)
at Dnn.ExportImport.Components.Common.CompressionUtil.AddFileToArchive(String file, String archivePath, Int32 folderOffset, String folder)
at Dnn.ExportImport.Components.Services.PagesExportService.ExportModulePackage(ExportModule exportModule)
at Dnn.ExportImport.Components.Services.PagesExportService.ExportTabModulesAndRelatedItems(ExportTab exportPage, DateTime toDate, Nullable`1 fromDate)
at Dnn.ExportImport.Components.Services.PagesExportService.ProcessExportPages()
at Dnn.ExportImport.Components.Services.PagesExportService.ExportData(ExportImportJob exportJob, ExportDto exportDto)
at Dnn.ExportImport.Components.Engines.ExportImportEngine.Export(ExportImportJob exportJob, ExportImportResult result, ScheduleHistoryItem scheduleHistoryItem)
at Dnn.ExportImport.Components.Scheduler.ExportImportScheduler.DoWork()
```
## Expected behavior
Export should finish successfully.

## Affected version
* [X] 10.00.00 alpha build
* [X] 09.06.02 release candidate
* [X] 09.06.01 latest supported release

## Affected browser
* [X] Chrome
* [X] Firefox
* [X] Safari
* [X] Internet Explorer 11
* [X] Microsoft Edge (Classic)
* [X] Microsoft Edge Chromium

--- a/DNN Platform/Modules/DnnExportImport/Components/Common/CompressionUtil.cs
+++ b/DNN Platform/Modules/DnnExportImport/Components/Common/CompressionUtil.cs
@@ -12,6 +12,7 @@ namespace Dnn.ExportImport.Components.Common
     using System.IO.MemoryMappedFiles;
     using System.Linq;
     using System.Text;
+    using DotNetNuke.Common.Utilities.Internal;
 
     public static class CompressionUtil
     {
@@ -199,6 +200,27 @@ namespace Dnn.ExportImport.Components.Common
         /// <param name="archiveFileName"></param>
         /// <returns></returns>
         public static ZipArchive OpenCreate(string archiveFileName)
+        {
+            if (string.IsNullOrWhiteSpace(archiveFileName))
+            {
+                throw new ArgumentNullException(nameof(archiveFileName));
+            }
+
+            ZipArchive zip = null;
+
+            RetryableAction.RetryEverySecondFor30Seconds(
+                () => zip = OpenCreateUnsafe(archiveFileName),
+                $"{nameof(OpenCreateUnsafe)}(\"{archiveFileName}\")");
+
+            return zip;
+        }
+
+        /// <summary>
+        /// Open the archive file for read and write (no retry).
+        /// </summary>
+        /// <param name="archiveFileName">The full zip file path.</param>
+        /// <returns>A <see cref="ZipArchive"/> instance.</returns>
+        internal static ZipArchive OpenCreateUnsafe(string archiveFileName)
         {
             return File.Exists(archiveFileName)
                 ? ZipFile.Open(archiveFileName, ZipArchiveMode.Update, Encoding.UTF8)

